src/Slices/ServiceInstanceDetails/UI/Tabs/ResourcesTabContent.tsx

```diff
@@ -1,10 +1,116 @@
-import React from "react";
-import { TabContent } from "@patternfly/react-core";
+import React, { useContext } from "react";
+import {
+  Content,
+  DescriptionList,
+  DescriptionListDescription,
+  DescriptionListGroup,
+  DescriptionListTerm,
+  Flex,
+  FlexItem,
+  TabContent,
+} from "@patternfly/react-core";
+import { DeploymentProgress } from "@/Core";
+import { useGetInstanceResources } from "@/Data/Managers/V2/GETTERS/GetInstanceResources";
+import { DeploymentProgressBar } from "@/Slices/ServiceInventory/UI/Components";
+import { DependencyContext, words } from "@/UI";
+import {
+  EmptyView,
+  ErrorView,
+  LoadingView,
+  ResourceTable,
+} from "@/UI/Components";
+import { InstanceDetailsContext } from "../../Core/Context";
 
 export const ResourcesTabContent: React.FC = () => {
+  const { instance } = useContext(InstanceDetailsContext);
+  const { environmentHandler } = useContext(DependencyContext);
+  const environment = environmentHandler.useId();
+
+  const { data, isSuccess, isError, error } = useGetInstanceResources(
+    instance.id,
+    instance.service_entity,
+    String(instance.version),
+    environment,
+  ).useContinuous();
+
+  if (isSuccess) {
+    return (
+      <TabContent role="tabpanel" id={"Resources-content"}>
+        {data.length === 0 ? (
+          <EmptyView
+            title={words("inventory.resourcesTab.empty.title")}
+            message={words("inventory.resourcesTab.empty.body")}
+          />
+        ) : (
+          <>
+            <DeploymentProgress
+              deployment_progress={instance.deployment_progress}
+            />
+            <ResourceTable resources={data} aria-label="Resource-table" />
+          </>
+        )}
+      </TabContent>
+    );
+  }
+
+  if (isError) {
+    return (
+      <TabContent role="tabpanel" id={"Resources-content"}>
+        <ErrorView
+          message={error.message}
+          ariaLabel="Error_view-Resources-content"
+        />
+      </TabContent>
+    );
+  }
+
   return (
     <TabContent role="tabpanel" id={"Resources-content"}>
-      Temporary Resources Content
+      <LoadingView ariaLabel="Resources-Loading" />
     </TabContent>
   );
 };
+
+interface Props {
+  deployment_progress?: DeploymentProgress | null;
+}
+
+const DeploymentProgress: React.FC<Props> = ({ deployment_progress }) => {
+  if (!deployment_progress) {
+    return (
+      <Flex>{words("instanceDetails.tabs.resources.EmptyResources")}</Flex>
+    );
+  }
+
+  return (
+    <DescriptionList
+      isHorizontal
+      horizontalTermWidthModifier={{
+        default: "15ch",
+      }}
+      style={{
+        marginTop: "var(--pf-t--global--spacer--300)",
```

Comment:
they have specific spacers for different purposes, we should avoid using the primitive value, and use the variables that are specific. But I think you might not need them if you use the tab-wrapper component mentioned in a previous comment, or the TabContentBody. At least, it would solve the top margin being required.

I see I used --pf-t--global--spacer--action--horizontal--spacious in the wrapper component, for the top spacing

Reply:
TabContentWrapper solved top margin, bodyContent would require overflow, so I opted for your wrapper anyway

Btw. for your wrapper`--vertical--`  should be better fit ;D